src/panels/lovelace/badges/hui-state-label-badge.ts

```diff
@@ -44,26 +46,17 @@ export class HuiStateLabelBadge extends LitElement implements LovelaceBadge {
           : ""}
         .icon=${this._config.icon}
         .image=${this._config.image}
-        @ha-click=${this._handleClick}
-        @ha-hold=${this._handleHold}
-        @ha-dblclick=${this._handleDblClick}
-        .longPress=${longPress({
-          hasDoubleClick: hasDoubleClick(this._config!.double_tap_action),
+        @action=${this._handleAction}
+        .actionHandler=${actionHandler({
+          hasHold: hasAction(this._config!.hold_action),
+          hasDoubleClick: hasAction(this._config!.double_tap_action),
         })}
       ></ha-state-label-badge>
     `;
   }
 
-  private _handleClick() {
-    handleClick(this, this.hass!, this._config!, false, false);
-  }
-
-  private _handleHold() {
-    handleClick(this, this.hass!, this._config!, true, false);
-  }
-
-  private _handleDblClick() {
-    handleClick(this, this.hass!, this._config!, false, true);
+  private _handleAction(ev: HASSDomEvent<ActionHandlerEvent>) {
```

Comment:
Instead of importing `HASSDomEvent` in every file, you also export `ActionHandlerEvent` with `HASSDomEvent`:

```ts
export interface ActionHandlerDetail {
  action?: string;
}

export type ActionHandlerEvent = HASSDomEvent<ActionHandlerDetail>;
```